src/Core/Src/app_threadx.c

```diff
@@ -63,6 +64,11 @@ UINT App_ThreadX_Init(VOID *memory_ptr)
 
   /* USER CODE BEGIN App_ThreadX_Init */
   (void)byte_pool;
+
+  ret = queue_send_thread_create(byte_pool);
+
+  ret = queue_receive_thread_create(byte_pool);
```

Comment:
Should only be calling the second function if `ret == TX_SUCCESS`, otherwise the first could fail and the second succeed meaning the error return code would be overwritten.

Reply:
This is definitely better. Automotive embedded C coding standards (specifically, MISRA C 2012 which I try to follow as much as possible), advise to avoid having "early returns" in a function. The motivation for having a single point of return  is to avoid the possibility of returning and forgetting to clean up resources. I find that this can reduce readability, but I've still followed it in other projects because dynamic memory allocation services are used in them. It's up to you whether you do or not, but just be careful of the chance of forgetting to clean up resources if you start using any ThreadX dynamic memory allocation services.